src/NuGet.Core/NuGet.Protocol/Utility/MetadataReferenceCache.cs

```diff
@@ -17,6 +17,8 @@ public class MetadataReferenceCache
         private readonly Dictionary<string, string> _stringCache = new Dictionary<string, string>(StringComparer.Ordinal);
         private readonly Dictionary<Type, PropertyInfo[]> _propertyCache = new Dictionary<Type, PropertyInfo[]>();
         private readonly Dictionary<string, NuGetVersion> _versionCache = new Dictionary<string, NuGetVersion>(StringComparer.Ordinal);
+        private readonly Dictionary<Type, MethodInfo> _stringMethodsCache = new Dictionary<Type, MethodInfo>();
+        private readonly Type _metadataReferenceCacheType = typeof(MetadataReferenceCache);
```

Comment:
This class is `MetadataReferenceCache`. It seems very strange to me why you'd need to use reflection for this class to access itself.

Reply:
I didn't add it it was there actually before my change, I just moved to top for multiple use.
![image](https://user-images.githubusercontent.com/8766776/85607861-5e395d00-b609-11ea-8c5a-3cd29a652efc.png)